enhancements/ingress/add-dns-and-loadbalancer-conditions-to-managed-gateway.md

```diff
@@ -0,0 +1,621 @@
+
+**Upgrade Strategy:**
+* This is a purely additive feature with no configuration changes required
+* On upgrade to a version with this enhancement:
+  - Existing Gateway resources in `openshift-ingress` namespace will have 
+  conditions added during next reconciliation
+  - No user action required
+  - No disruption to existing Gateways or traffic flow
+* The feature is backwards compatible; older clients that don't understand 
+these conditions will ignore them
+
+**Downgrade Strategy:**
+* On downgrade to a version without this enhancement:
+  - The new conditions will remain in the Gateway status but will not be updated
+  - The CVO does not delete status fields, so conditions will persist but become stale
+  - No impact on Gateway functionality; conditions are read-only status
+  - If desired, users can manually remove the conditions, but this is not required
+
+**Version Skew:**
+* Ingress operator and Gateway resources can be at different versions during upgrade
+* Older ingress operator will not add/update the new conditions (graceful degradation)
+* Newer ingress operator will add conditions to any Gateway in `openshift-ingress` regardless of when it was created
+
+## Version Skew Strategy
+
+This enhancement involves coordination between:
+* Ingress operator (adds conditions to Gateway status)
+* Gateway API controller (adds labels, creates services)
+
+**Version Skew Handling:**
+* No kubelet changes are involved
+* No CSI, CRI, or CNI changes are involved
+
+**Compatibility:**
+* Feature works with Gateway API v1 (both support custom conditions)
+* No breaking changes to Gateway API contract
+* Follows standard Kubernetes condition conventions
+
+## Operational Aspects of API Extensions
+
+**API Extension Type:**
+* This enhancement adds status conditions to existing Gateway resources (no new CRDs)
+* No admission webhooks, conversion webhooks, aggregated API servers, or finalizers are introduced
+* Status conditions are a standard Kubernetes pattern and do not require API approval for new types
+
+**Impact on Existing SLIs:**
+* Negligible impact on API throughput: condition updates happen during normal reconciliation
+* No new API calls introduced; only status updates to existing Gateway resources
+* Expected number of managed Gateways in `openshift-ingress` namespace: typically 1-10 per cluster
+* Condition updates are rate-limited to prevent excessive writes
+
+**Failure Modes:**
+* If ingress operator is unavailable, conditions will not be updated, but Gateways continue to function
+* Conditions are status-only and do not affect Gateway data plane functionality
+
+**Health Indicators:**
+* Monitor ingress-operator pod health and logs for errors updating Gateway conditions
+* Existing ingress-operator metrics and alerts apply
+
+**Responsible Teams:**
+* Networking team (ingress): primary owner for Gateway condition logic
+
+## Support Procedures
+
+**Detecting Issues:**
+
+*Symptom: Gateway conditions show `DNSManaged=False`*
```

Comment:
This is not necessarily an error condition.  Some users may choose to not have DNS be managed.